mmedit/datasets/pipelines/matlab_resize.py

```diff
@@ -0,0 +1,270 @@
+from math import ceil
+
+import numpy as np
+
+from ..registry import PIPELINES
+
+
+def get_size_from_scale(input_size, scale_factor):
+    """Get the output size given input size and scale factor.
+
+    Args:
+        input_size (tuple): The size of the input image.
+        scale_factor (float): The resize factor.
+
+    Returns:
+        list[int]: The size of the output image.
+    """
+
+    output_shape = [
+        int(ceil(scale * shape))
+        for (scale, shape) in zip(scale_factor, input_size)
+    ]
+
+    return output_shape
+
+
+def get_scale_from_size(input_size, output_size):
+    """Get the scale factor given input size and output size.
+
+    Args:
+        input_size (tuple(int)): The size of the input image.
+        output_size (tuple(int)): The size of the output image.
+
+    Returns:
+        list[float]: The scale factor of each dimension.
+    """
+
+    scale = [
+        1.0 * output_shape / input_shape
+        for (input_shape, output_shape) in zip(input_size, output_size)
+    ]
+
+    return scale
+
+
+def cubic(x):
+    """ Cubic function.
+
+    Args:
+        x (ndarray): The distance from the center position.
+
+    Returns:
+        ndarray: The weight corresponding to a particular distance.
+
+    """
+
+    x = np.array(x).astype(np.float32)
+    x_abs = np.absolute(x)
+    x_abs_sq = np.multiply(x_abs, x_abs)
+    x_abs_cu = np.multiply(x_abs_sq, x_abs)
+    f = np.multiply(1.5 * x_abs_cu - 2.5 * x_abs_sq + 1,
+                    x_abs <= 1) + np.multiply(
+                        -0.5 * x_abs_cu + 2.5 * x_abs_sq - 4 * x_abs + 2,
+                        (1 < x_abs) & (x_abs <= 2))
+
+    return f
+
+
+def get_weights_indices(input_length, output_length, scale, kernel,
+                        kernel_width):
+    """Get weights and indices for interpolation.
+
+    Args:
+        input_length (int): Length of the input sequence.
+        output_length (int): Length of the output sequence.
+        scale (float): Scale factor.
+        kernel (func): The kernel used for resizing.
+        kernel_width (int): The width of the kernel.
+
+    Returns:
+        list[ndarray]: The weights and the indices for interpolation.
+
+
+    """
+    if scale < 1:  # modified kernel for antialiasing
+
+        def h(x):
+            return scale * kernel(scale * x)
+
+        kernel_width = 1.0 * kernel_width / scale
+    else:
+        h = kernel
+        kernel_width = kernel_width
+
+    # coordinates of output
+    x = np.arange(1, output_length + 1).astype(np.float32)
+
+    # coordinates of input
+    u = x / scale + 0.5 * (1 - 1 / scale)
+    left = np.floor(u - kernel_width / 2)  # leftmost pixel
+    p = int(ceil(kernel_width)) + 2  # maximum number of pixels
+
+    # indices of input pixels
+    ind = np.expand_dims(left, axis=1) + np.arange(p) - 1
+    indices = ind.astype(np.int32)
+
+    # weights of input pixels
+    weights = h(np.expand_dims(u, axis=1) - indices - 1)
+    weights = np.divide(weights,
```

Comment:
same here, 

* use `/` to replace np.divide
* use None to replace np.expand_dims